optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/DotNames.java

```diff
@@ -18,20 +18,44 @@
 
 import org.jboss.jandex.DotName;
 import org.optaplanner.core.api.domain.entity.PlanningEntity;
+import org.optaplanner.core.api.domain.entity.PlanningPin;
+import org.optaplanner.core.api.domain.lookup.PlanningId;
 import org.optaplanner.core.api.domain.solution.PlanningSolution;
+import org.optaplanner.core.api.domain.variable.AnchorShadowVariable;
+import org.optaplanner.core.api.domain.variable.CustomShadowVariable;
+import org.optaplanner.core.api.domain.variable.InverseRelationShadowVariable;
+import org.optaplanner.core.api.domain.variable.PlanningVariable;
 import org.optaplanner.core.api.score.calculator.EasyScoreCalculator;
 import org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator;
 import org.optaplanner.core.api.score.stream.ConstraintProvider;
 
 public final class DotNames {
 
     static final DotName PLANNING_SOLUTION = DotName.createSimple(PlanningSolution.class.getName());
-    static final DotName PLANNING_ENTITY = DotName.createSimple(PlanningEntity.class.getName());
     static final DotName EASY_SCORE_CALCULATOR = DotName.createSimple(EasyScoreCalculator.class.getName());
     static final DotName CONSTRAINT_PROVIDER = DotName.createSimple(ConstraintProvider.class.getName());
     static final DotName INCREMENTAL_SCORE_CALCULATOR =
             DotName.createSimple(IncrementalScoreCalculator.class.getName());
 
+    static final DotName PLANNING_ENTITY = DotName.createSimple(PlanningEntity.class.getName());
+    static final DotName PLANNING_PIN = DotName.createSimple(PlanningPin.class.getName());
+    static final DotName PLANNING_ID = DotName.createSimple(PlanningId.class.getName());
+
+    static final DotName PLANNING_VARIABLE = DotName.createSimple(PlanningVariable.class.getName());
+
+    static final DotName ANCHOR_SHADOW_VARIABLE = DotName.createSimple(AnchorShadowVariable.class.getName());
+    static final DotName CUSTOM_SHADOW_VARIABLE = DotName.createSimple(CustomShadowVariable.class.getName());
+    static final DotName INVERSE_RELATION_SHADOW_VARIABLE = DotName.createSimple(InverseRelationShadowVariable.class.getName());
```

Comment:
Nitpick:
The order should be:
- domain then scoring, so EASY_SCORE_CALCULATOR, CONSTRAINT_PROVIDER and INCREMENTAL_SCORE_CALCULATOR last
- entity class, variables (genuine, shadow), id, pin
- in shadow vars part: custom last. anchor depends on inverse, so inverse before anchor.

Reply:
This PR only added the DotNames needed for this feature, which is a strict subset of those added in the GizmoMemberAccessor (where it was argued that the DotNames should roughly follow the processing order of annotations). Given that the GizmoMemberAccessor PR is merged, I am not sure if the comment still applies